packages/core/core/src/services/document-service/document-engine.ts

```diff
@@ -56,44 +62,59 @@ const createDocumentEngine = ({
   async findMany(uid, params) {
     const { kind } = strapi.getModel(uid);
 
-    const query = transformParamsToQuery(uid, params || ({} as any));
-    query.where = { ...params?.lookup, ...query.where };
+    const query = await pipeAsync(
+      (params) => transformParamsDocumentId(uid, params, { isDraft: true, locale: params.locale }),
+      (params) => transformParamsToQuery(uid, params),
+      (query) => set('where', { ...params?.lookup, ...query.where }, query)
+    )(params || {});
 
     if (kind === 'singleType') {
       return db.query(uid).findOne(query);
     }
 
-    return db.query(uid).findMany(query);
+    return db
+      .query(uid)
+      .findMany(query)
+      .then((doc) => transformOutputDocumentId(uid, doc));
   },
 
   async findFirst(uid, params) {
-    const query = transformParamsToQuery(uid, params || ({} as any));
+    const query = await pipeAsync(
+      (params) => transformParamsDocumentId(uid, params, { isDraft: true, locale: params.locale }),
+      (params) => transformParamsToQuery(uid, params)
+    )(params || {});
 
-    return db.query(uid).findOne({ ...query, where: { ...params?.lookup, ...query.where } });
+    return db
+      .query(uid)
+      .findOne({ ...query, where: { ...params?.lookup, ...query.where } })
+      .then((doc) => transformOutputDocumentId(uid, doc));
   },
 
   async findOne(uid, documentId, params) {
-    const query = transformParamsToQuery(uid, params || ({} as any));
+    const query = await pipeAsync(
+      (params) => transformParamsDocumentId(uid, params, { isDraft: true, locale: params.locale }),
+      (params) => transformParamsToQuery(uid, params)
+    )(params || {});
+
     return db
       .query(uid)
-      .findOne({ ...query, where: { ...params?.lookup, ...query.where, documentId } });
+      .findOne({ ...query, where: { ...params?.lookup, ...query.where, documentId } })
+      .then((doc) => transformOutputDocumentId(uid, doc));
   },
 
   async delete(uid, documentId, params = {} as any) {
-    const query = transformParamsToQuery(uid, params as any);
+    const query = await pipeAsync(
+      // TODO: What if we are deleting more than one locale / publication state?
```

Comment:
How does delete differ from deleteMany if both can delete multiple entries?

Reply:
on the delete you shold provide a document id, and you can delete multiple of its locales or publication states,
wether on the deleteMany you can delete any document